Currently, `nix run .#makeAll` fails with this log:

```
make -C arch BUILD_DIR=/home/eyjafjallajokull/Desktop/gh/alias-c/build
make[1]: Entering directory '/home/eyjafjallajokull/Desktop/gh/alias-c/arch'
mkdir -p /home/eyjafjallajokull/Desktop/gh/alias-c/build/arch
nasm -f elf64 x86/memory.asm -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/arch/memory.o
nasm -f elf64 x86/posix.asm -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/arch/posix.o
make[1]: Leaving directory '/home/eyjafjallajokull/Desktop/gh/alias-c/arch'
make -C stdlib BUILD_DIR=/home/eyjafjallajokull/Desktop/gh/alias-c/build
make[1]: Entering directory '/home/eyjafjallajokull/Desktop/gh/alias-c/stdlib'
mkdir -p /home/eyjafjallajokull/Desktop/gh/alias-c/build/stdlib
gcc -g -fno-stack-protector -I include -c src/algorithm.c -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/stdlib/algorithm.o
gcc -g -fno-stack-protector -I include -c src/cassert.c -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/stdlib/cassert.o
gcc -g -fno-stack-protector -I include -c src/heap.c -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/stdlib/heap.o
gcc -g -fno-stack-protector -I include -c src/heap_test.c -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/stdlib/heap_test.o
gcc -g -fno-stack-protector -I include -c src/ordered_array.c -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/stdlib/ordered_array.o
gcc -g -fno-stack-protector -I include -c src/panic.c -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/stdlib/panic.o
gcc -g -fno-stack-protector -I include -c src/stdio.c -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/stdlib/stdio.o
gcc -g -fno-stack-protector -I include -c src/stdlib.c -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/stdlib/stdlib.o
gcc -g -fno-stack-protector -I include -c src/string.c -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/stdlib/string.o
gcc -g -fno-stack-protector -I include -c src/vector.c -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/stdlib/vector.o
make[1]: Leaving directory '/home/eyjafjallajokull/Desktop/gh/alias-c/stdlib'
make -C compiler BUILD_DIR=/home/eyjafjallajokull/Desktop/gh/alias-c/build
make[1]: Entering directory '/home/eyjafjallajokull/Desktop/gh/alias-c/compiler'
mkdir -p /home/eyjafjallajokull/Desktop/gh/alias-c/build/compiler
gcc -g -fno-stack-protector -I include -I ../stdlib/include -c src/compile.c -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/compiler/compile.o
gcc -g -fno-stack-protector -I include -I ../stdlib/include -c src/context.c -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/compiler/context.o
gcc -g -fno-stack-protector -I include -I ../stdlib/include -c src/exception.c -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/compiler/exception.o
gcc -g -fno-stack-protector -I include -I ../stdlib/include -c src/languageserver.c -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/compiler/languageserver.o
gcc -g -fno-stack-protector -I include -I ../stdlib/include -c src/lexer.c -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/compiler/lexer.o
gcc -g -fno-stack-protector -I include -I ../stdlib/include -c src/main.c -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/compiler/main.o
gcc -g -fno-stack-protector -I include -I ../stdlib/include -c src/process.c -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/compiler/process.o
gcc -g -fno-stack-protector -I include -I ../stdlib/include -c src/syntax.c -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/compiler/syntax.o
gcc -g -fno-stack-protector -I include -I ../stdlib/include -c src/token.c -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/compiler/token.o
gcc -g -fno-stack-protector -I include -I ../stdlib/include -c src/type.c -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/compiler/type.o
gcc -g -fno-stack-protector -I include -c httpd/httpd.c -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/compiler/httpd.o
gcc -z noexecstack -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/calias  /home/eyjafjallajokull/Desktop/gh/alias-c/build/compiler/compile.o  /home/eyjafjallajokull/Desktop/gh/alias-c/build/compiler/context.o  /home/eyjafjallajokull/Desktop/gh/alias-c/build/compiler/exception.o  /home/eyjafjallajokull/Desktop/gh/alias-c/build/compiler/languageserver.o  /home/eyjafjallajokull/Desktop/gh/alias-c/build/compiler/lexer.o  /home/eyjafjallajokull/Desktop/gh/alias-c/build/compiler/main.o  /home/eyjafjallajokull/Desktop/gh/alias-c/build/compiler/process.o  /home/eyjafjallajokull/Desktop/gh/alias-c/build/compiler/syntax.o  /home/eyjafjallajokull/Desktop/gh/alias-c/build/compiler/token.o  /home/eyjafjallajokull/Desktop/gh/alias-c/build/compiler/type.o /home/eyjafjallajokull/Desktop/gh/alias-c/build/compiler/httpd.o /home/eyjafjallajokull/Desktop/gh/alias-c/build/arch/memory.o /home/eyjafjallajokull/Desktop/gh/alias-c/build/arch/posix.o /home/eyjafjallajokull/Desktop/gh/alias-c/build/stdlib/algorithm.o /home/eyjafjallajokull/Desktop/gh/alias-c/build/stdlib/cassert.o /home/eyjafjallajokull/Desktop/gh/alias-c/build/stdlib/heap.o /home/eyjafjallajokull/Desktop/gh/alias-c/build/stdlib/heap_test.o /home/eyjafjallajokull/Desktop/gh/alias-c/build/stdlib/ordered_array.o /home/eyjafjallajokull/Desktop/gh/alias-c/build/stdlib/panic.o /home/eyjafjallajokull/Desktop/gh/alias-c/build/stdlib/stdio.o /home/eyjafjallajokull/Desktop/gh/alias-c/build/stdlib/stdlib.o /home/eyjafjallajokull/Desktop/gh/alias-c/build/stdlib/string.o /home/eyjafjallajokull/Desktop/gh/alias-c/build/stdlib/vector.o
make[1]: Leaving directory '/home/eyjafjallajokull/Desktop/gh/alias-c/compiler'
make -C arch BUILD_DIR=/home/eyjafjallajokull/Desktop/gh/alias-c/build
make[1]: Entering directory '/home/eyjafjallajokull/Desktop/gh/alias-c/arch'
mkdir -p /home/eyjafjallajokull/Desktop/gh/alias-c/build/arch
nasm -f elf64 x86/memory.asm -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/arch/memory.o
nasm -f elf64 x86/posix.asm -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/arch/posix.o
make[1]: Leaving directory '/home/eyjafjallajokull/Desktop/gh/alias-c/arch'
make -C stdlib BUILD_DIR=/home/eyjafjallajokull/Desktop/gh/alias-c/build
make[1]: Entering directory '/home/eyjafjallajokull/Desktop/gh/alias-c/stdlib'
mkdir -p /home/eyjafjallajokull/Desktop/gh/alias-c/build/stdlib
gcc -g -fno-stack-protector -I include -c src/algorithm.c -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/stdlib/algorithm.o
gcc -g -fno-stack-protector -I include -c src/cassert.c -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/stdlib/cassert.o
gcc -g -fno-stack-protector -I include -c src/heap.c -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/stdlib/heap.o
gcc -g -fno-stack-protector -I include -c src/heap_test.c -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/stdlib/heap_test.o
gcc -g -fno-stack-protector -I include -c src/ordered_array.c -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/stdlib/ordered_array.o
gcc -g -fno-stack-protector -I include -c src/panic.c -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/stdlib/panic.o
gcc -g -fno-stack-protector -I include -c src/stdio.c -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/stdlib/stdio.o
gcc -g -fno-stack-protector -I include -c src/stdlib.c -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/stdlib/stdlib.o
gcc -g -fno-stack-protector -I include -c src/string.c -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/stdlib/string.o
gcc -g -fno-stack-protector -I include -c src/vector.c -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/stdlib/vector.o
make[1]: Leaving directory '/home/eyjafjallajokull/Desktop/gh/alias-c/stdlib'
make -C compiler BUILD_DIR=/home/eyjafjallajokull/Desktop/gh/alias-c/build
make[1]: Entering directory '/home/eyjafjallajokull/Desktop/gh/alias-c/compiler'
mkdir -p /home/eyjafjallajokull/Desktop/gh/alias-c/build/compiler
gcc -g -fno-stack-protector -I include -I ../stdlib/include -c src/compile.c -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/compiler/compile.o
gcc -g -fno-stack-protector -I include -I ../stdlib/include -c src/context.c -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/compiler/context.o
gcc -g -fno-stack-protector -I include -I ../stdlib/include -c src/exception.c -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/compiler/exception.o
gcc -g -fno-stack-protector -I include -I ../stdlib/include -c src/languageserver.c -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/compiler/languageserver.o
gcc -g -fno-stack-protector -I include -I ../stdlib/include -c src/lexer.c -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/compiler/lexer.o
gcc -g -fno-stack-protector -I include -I ../stdlib/include -c src/main.c -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/compiler/main.o
gcc -g -fno-stack-protector -I include -I ../stdlib/include -c src/process.c -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/compiler/process.o
gcc -g -fno-stack-protector -I include -I ../stdlib/include -c src/syntax.c -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/compiler/syntax.o
gcc -g -fno-stack-protector -I include -I ../stdlib/include -c src/token.c -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/compiler/token.o
gcc -g -fno-stack-protector -I include -I ../stdlib/include -c src/type.c -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/compiler/type.o
gcc -g -fno-stack-protector -I include -c httpd/httpd.c -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/compiler/httpd.o
gcc -z noexecstack -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/calias  /home/eyjafjallajokull/Desktop/gh/alias-c/build/compiler/compile.o  /home/eyjafjallajokull/Desktop/gh/alias-c/build/compiler/context.o  /home/eyjafjallajokull/Desktop/gh/alias-c/build/compiler/exception.o  /home/eyjafjallajokull/Desktop/gh/alias-c/build/compiler/languageserver.o  /home/eyjafjallajokull/Desktop/gh/alias-c/build/compiler/lexer.o  /home/eyjafjallajokull/Desktop/gh/alias-c/build/compiler/main.o  /home/eyjafjallajokull/Desktop/gh/alias-c/build/compiler/process.o  /home/eyjafjallajokull/Desktop/gh/alias-c/build/compiler/syntax.o  /home/eyjafjallajokull/Desktop/gh/alias-c/build/compiler/token.o  /home/eyjafjallajokull/Desktop/gh/alias-c/build/compiler/type.o /home/eyjafjallajokull/Desktop/gh/alias-c/build/compiler/httpd.o /home/eyjafjallajokull/Desktop/gh/alias-c/build/arch/memory.o /home/eyjafjallajokull/Desktop/gh/alias-c/build/arch/posix.o /home/eyjafjallajokull/Desktop/gh/alias-c/build/stdlib/algorithm.o /home/eyjafjallajokull/Desktop/gh/alias-c/build/stdlib/cassert.o /home/eyjafjallajokull/Desktop/gh/alias-c/build/stdlib/heap.o /home/eyjafjallajokull/Desktop/gh/alias-c/build/stdlib/heap_test.o /home/eyjafjallajokull/Desktop/gh/alias-c/build/stdlib/ordered_array.o /home/eyjafjallajokull/Desktop/gh/alias-c/build/stdlib/panic.o /home/eyjafjallajokull/Desktop/gh/alias-c/build/stdlib/stdio.o /home/eyjafjallajokull/Desktop/gh/alias-c/build/stdlib/stdlib.o /home/eyjafjallajokull/Desktop/gh/alias-c/build/stdlib/string.o /home/eyjafjallajokull/Desktop/gh/alias-c/build/stdlib/vector.o
make[1]: Leaving directory '/home/eyjafjallajokull/Desktop/gh/alias-c/compiler'
make -C altlib BUILD_DIR=/home/eyjafjallajokull/Desktop/gh/alias-c/build
make[1]: Entering directory '/home/eyjafjallajokull/Desktop/gh/alias-c/altlib'
mkdir -p /home/eyjafjallajokull/Desktop/gh/alias-c/build/altlib
mkdir -p /home/eyjafjallajokull/Desktop/gh/alias-c/build/altlib_ext/include
mkdir -p /home/eyjafjallajokull/Desktop/gh/alias-c/build/altlib_ext/lib
/home/eyjafjallajokull/Desktop/gh/alias-c/build/calias -a -i altlib ./ algorithm.al -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/altlib/algorithm.o
/home/eyjafjallajokull/Desktop/gh/alias-c/build/calias -a -i altlib ./ cassert.al -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/altlib/cassert.o
/home/eyjafjallajokull/Desktop/gh/alias-c/build/calias -a -i altlib ./ memory.al -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/altlib/memory.o
/home/eyjafjallajokull/Desktop/gh/alias-c/build/calias -a -i altlib ./ posix.al -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/altlib/posix.o
/home/eyjafjallajokull/Desktop/gh/alias-c/build/calias -a -i altlib ./ stdio.al -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/altlib/stdio.o
/home/eyjafjallajokull/Desktop/gh/alias-c/build/calias -a -i altlib ./ stdlib.al -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/altlib/stdlib.o
/home/eyjafjallajokull/Desktop/gh/alias-c/build/calias -a -i altlib ./ string.al -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/altlib/string.o
/home/eyjafjallajokull/Desktop/gh/alias-c/build/calias -a -i altlib ./ test_allocator.al -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/altlib/test_allocator.o
/home/eyjafjallajokull/Desktop/gh/alias-c/build/calias -a -i altlib ./ vector.al -o /home/eyjafjallajokull/Desktop/gh/alias-c/build/altlib/vector.o
cat header.html > ../docs/altlibref.html
/home/eyjafjallajokull/Desktop/gh/alias-c/build/altlib/generate_docs algorithm.al cassert.al memory.al posix.al stdio.al stdlib.al string.al test_allocator.al vector.al >> ../docs/altlibref.html
cp algorithm.al cassert.al memory.al posix.al stdio.al stdlib.al string.al test_allocator.al vector.al /home/eyjafjallajokull/Desktop/gh/alias-c/build/altlib_ext/include
cp  /home/eyjafjallajokull/Desktop/gh/alias-c/build/altlib/algorithm.o  /home/eyjafjallajokull/Desktop/gh/alias-c/build/altlib/cassert.o  /home/eyjafjallajokull/Desktop/gh/alias-c/build/altlib/memory.o  /home/eyjafjallajokull/Desktop/gh/alias-c/build/altlib/posix.o  /home/eyjafjallajokull/Desktop/gh/alias-c/build/altlib/stdio.o  /home/eyjafjallajokull/Desktop/gh/alias-c/build/altlib/stdlib.o  /home/eyjafjallajokull/Desktop/gh/alias-c/build/altlib/string.o  /home/eyjafjallajokull/Desktop/gh/alias-c/build/altlib/test_allocator.o  /home/eyjafjallajokull/Desktop/gh/alias-c/build/altlib/vector.o /home/eyjafjallajokull/Desktop/gh/alias-c/build/altlib_ext/lib
cp: cannot stat '/home/eyjafjallajokull/Desktop/gh/alias-c/build/altlib/algorithm.o': No such file or directory
cp: cannot stat '/home/eyjafjallajokull/Desktop/gh/alias-c/build/altlib/cassert.o': No such file or directory
cp: cannot stat '/home/eyjafjallajokull/Desktop/gh/alias-c/build/altlib/memory.o': No such file or directory
cp: cannot stat '/home/eyjafjallajokull/Desktop/gh/alias-c/build/altlib/posix.o': No such file or directory
cp: cannot stat '/home/eyjafjallajokull/Desktop/gh/alias-c/build/altlib/stdio.o': No such file or directory
cp: cannot stat '/home/eyjafjallajokull/Desktop/gh/alias-c/build/altlib/stdlib.o': No such file or directory
cp: cannot stat '/home/eyjafjallajokull/Desktop/gh/alias-c/build/altlib/string.o': No such file or directory
cp: cannot stat '/home/eyjafjallajokull/Desktop/gh/alias-c/build/altlib/test_allocator.o': No such file or directory
cp: cannot stat '/home/eyjafjallajokull/Desktop/gh/alias-c/build/altlib/vector.o': No such file or directory
make[1]: *** [Makefile:12: all] Error 1
make[1]: Leaving directory '/home/eyjafjallajokull/Desktop/gh/alias-c/altlib'
make: *** [Makefile:15: altlib] Error 2
```

--- a/flake.nix
+++ b/flake.nix
@@ -23,6 +23,20 @@
               cp build/calias $out/bin/calias
             '';
           };
+
+          makeAll = pkgs.writeShellApplication {
+            name = "makeAll";
+            text = ''
+              make compiler
+              make altlib
+              make test
+              make perftest
+            '';
+            runtimeInputs = [
+              pkgs.nasm
+              pkgs.cmake
+            ];
+          };
         };
       in
       {

--- a/flake.nix
+++ b/flake.nix
@@ -14,6 +14,7 @@
         packages = {
           calias = pkgs.stdenv.mkDerivation {
             name = "calias";
+            meta.mainProgram = "calias";
             src = fs.toSource {
               root = ./.;
               fileset = fs.unions [
@@ -40,7 +41,6 @@
               cp -r build $out
               cp build/calias $out/bin/calias
             '';
-            meta.mainProgram = "calias";
           };
         };
       in
@@ -49,7 +49,6 @@
         devShells.default = pkgs.mkShell {
           buildInputs = [ packages.calias ];
           shellHook = ''
-            export CALIAS="${pkgs.lib.getExe packages.calias}"
             export ALTLIB="${packages.calias}/build/altlib_ext"
           '';
         };

